### PR TITLE
simulator: use tiup playground  to deploy `pd-simulator`

### LINKED ISSUE #5258: Make PD Simulator power.
## Development Task
In past, there are few pull requests for PD simulator, so it is outdated in the time. 
I want add some task to complete the simulator , it will be helpful for us to validate some the scheduler algorithm.

sub task:

- [ ] #5322
- [ ] #5259
- [x] #5288
- [ ] #5290
- [x] #5444

### LINKED ISSUE #5288: use tiup playground   to deploy `pd-simulator`
## Enhance
User can use tiup playground to deploy simulator.

### PR DESCRIPTION
**Pull request by bufferflies**
### What problem does this PR solve?

Issue Number: Close #5288

### What is changed and how does it work?

```commit-message
simulator: use tiup playground to deploy pd-simulator.
```

### Check List

Tests

- Manual test (add detailed scripts or steps below)
run cmd: 
 
`tiup playground v5.4.0 --host 172.16.4.10 --kv.binpath ./pd-simulator --kv=1 --db=0 --kv.config=./tikv.conf`

tikv config:

```
case-name="redundant-balance-region"
sim-tick-interval="1s"
store-io-per-second=100
```

Code changes

Side effects

Related changes

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by nolouch — COMMENTED** [review]
We default use `pull model`, `push model` must deploy a push gateway. do you run it?

**Comment by bufferflies** [comment]
> We default use `pull model`, `push model` must deploy a push gateway. do you run it?

yes, I have run it by manual.

File: tools/pd-simulator/main.go [hunk 1]
```diff
@@ -18,13 +18,16 @@ import (
 	"context"
 	"flag"
 	"fmt"
+
```
Comment (rleungx) [inline tools/pd-simulator/main.go:21]: remove the empty line

File: tools/pd-simulator/main.go [hunk 2]
```diff
@@ -104,6 +108,11 @@ func run(simCase string) {
 	}
 }
 
+func RunMetrics() {
+	http.Handle("/metrics", promhttp.Handler())
+	http.ListenAndServe(":20180", nil)
```
Comment (rleungx) [inline tools/pd-simulator/main.go:113]: How about using a random port?

Comment (bufferflies) [reply tools/pd-simulator/main.go]: using status port?

File: tools/pd-simulator/main.go (hunk 1 above)
Comment (bufferflies) [reply tools/pd-simulator/main.go]: fix

**Review by nolouch — APPROVED** [review]
lgtm

**Pull request merged**